kafka/handle.go

```diff
@@ -323,3 +336,72 @@ func (h *handle) setOAuthBearerTokenFailure(errstr string) error {
 	}
 	return newError(cErr)
 }
+
+// handleIOTrigger wraps up librdkafka's rd_kafka_queue_io_event_enable functionality to let Go be notified of
+// new events available on librdkafka queues. The idea is that we set up a pair of pipe FD's in startIOTrigger,
+// and ask librdkafka to write a byte to the pipe when we go from 0 -> 1 events in a queue (i.e. it's edge
```

Comment:
FWIW... 

My instinct is to prefer "papering over" the consumer polling than resorting to creating pipes. Can you use `rd_kafka_queue_yield` to exit the consumer poll early on done? The C# client has a consumer poll method that works like your producer.FlushContext. It doesn't bother calling queue_yield, but that would be possible using `CancellationToken.Register`. No one has complained about this yet - can you think of a practical scenario where that's not acceptable? I get that it's not appealing..

Is there any other way of waking up the go scheduler from C - even if that requires modifications to librdkafka?.

For blocking APIs that return a result (e.g. offsets for times) we really need some queue based alternatives...

Reply:
Hum. I reached for the IO triggering since it seemed like the reason it was in librdkafka at all was to support nodejs, and if you squint maybe the golang scheduler kind of looks like the nodejs event loop.

I must have missed rd_kafka_queue_yield, I didn't realise this was there at all. I have to spend a few hours looking more closely at the code, but I think that would perhaps lead to a less invasive PR - I wouldn't need to go and touch all the event handling stuff inside `eventPoll` for example.

Let me find some time to play around with rd_kafka_queue_yield and get back to you :)

Reply:
btw: we had a discussion about this and other contributions internally, hopefully we'll get a meeting organized to discuss.

Reply:
Yup just chatting with your people now, keen to do that. We're now carrying a number of patches in our fork, so it's definitely worth having a conversastion about which bits of functionality make sense to merge into confluent-kafka-go, what shape they have to be in, and some plan of attack for what order to do this in. 

I'll look into rd_kafka_queue_yield in the meanwhile but yeah let's have that broader conversation first before we start working on the detail here for sure.